server/homepage.py

```diff
@@ -0,0 +1,186 @@
+        return response
+
+    # Display information
+    cells = []
+
+    ##### Adam's Dynamic Cell Code for changing order remotely #####
+
+    # cell_options = HomeCellOrder.query.all()
+    # options = [
+    #     x.cell_type
+    #  for x in cell_options]
+
+    # print('options', options)
+
+    # for option in options:
+    #     if option == 'Laundry':
+    #         print('yay laundry', option)
+    #         laundryCell = get_top_laundry_cell(user).getCell()
+    #         cells.append(laundryCell)
+    #     elif option == 'Dining':
+    #         print('yay dining', option)
+    #         diningCell = get_popular_dining_cell(user).getCell()
+    #         cells.append(diningCell)
+    #     elif option == 'News':
+    #         newsCell = get_news_cell().getCell()
+    #         cells.append(newsCell)
+    #     elif option == 'Event':
+    #         if get_event_cell():
+    #             eventCell = get_event_cell().getCell()
+    #             cells.append(eventCell)
+    #     elif option == 'GSR':
+    #         gsrCell = get_study_spaces_cell().getCell()
+    #         cells.append(gsrCell)
+    #     else:
+    #         print('other', option)
+
+    laundry = get_top_laundry_cell(user)
+    dining = get_dining_cell(user)
+    news = get_news_cell()
+    gsr = get_study_spaces_cell()
+    calendar = get_university_event_cell()
+
+    if calendar is not None:
+        cells.append(calendar)
+
+    cells.append(dining)
+
+    if news is not None:
+        cells.append(news)
+    
+    cells.extend([gsr, laundry])
+
+    response = jsonify({"cells": [x.getCell() for x in cells]})
+    response.status_code = 200 # or 400 or whatever
+    return response
+
+# returns a dining cell containing the users preferences in sorted order
+def get_dining_preference_cell(user):
+    preferences = sqldb.session.query(DiningPreference.venue_id) \
+                               .filter_by(user_id=user.id).group_by(DiningPreference.venue_id) \
+                               .order_by(func.count(DiningPreference.venue_id).desc()).all()
+    venue_ids = [x.venue_id for x in preferences]
+    return HomeCell("dining", info)
+
+# returns a dining cell
+# TODO: personalize with preferences
+def get_dining_cell(user):
+    preferences = sqldb.session.query(DiningPreference.venue_id).filter_by(user_id=user.id)
+    venue_ids = [x.venue_id for x in preferences]
+    defaults_ids = [593, 1442, 636]
+    if len(venue_ids) == 0:
+        venue_ids = defaults_ids
+    elif len(venue_ids) == 1:
+        venue_ids.extend(defaults_ids)
+        venue_ids = list(set(venue_ids))[:3]
+
+    info = {"venues": venue_ids}
+    return HomeCell("dining", info)
+
+# returns a list of laundry cells
+def get_laundry_cells(user):
+    preferences = LaundryPreference.query.filter_by(user_id=user.id)
+    room_ids = [x.room_id for x in preferences]
+
+    # If the user has no preferences, select Bishop White
+    if len(room_ids) == 0:
```

Comment:
You can also do `if not room_ids:`. Arrays evaluate to false if they are empty, and true otherwise.